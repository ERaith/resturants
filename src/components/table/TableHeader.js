
export function TableHeader({ headers }) {

  return (
    <tr className="table-row">
      {
        headers.map((d) => <th key={d.key}>{d.text}</th>)
      }
    </tr>
  )
}
