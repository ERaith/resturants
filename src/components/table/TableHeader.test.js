import React from "react";
import { TableHeader } from "./TableHeader";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

describe("TableCell", () => {
  describe("Snapshots", () => {
    it("should match the snapshot", () => {
      const props = {
        headers: [
          {
            key: "name",
            text: "Resturant Name",
          },
          {
            key: "city",
            text: "CITY",
          },
        ],
      };
      const wrapper = mount(
        <table>
          <thead>
            <TableHeader {...props} />
          </thead>
        </table>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
