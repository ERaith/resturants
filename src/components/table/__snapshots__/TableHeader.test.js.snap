// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCell Snapshots should match the snapshot 1`] = `
<table>
  <thead>
    <TableHeader
      headers={
        Array [
          Object {
            "key": "name",
            "text": "Resturant Name",
          },
          Object {
            "key": "city",
            "text": "CITY",
          },
        ]
      }
    >
      <tr
        className="table-row"
      >
        <th
          key="name"
        >
          Resturant Name
        </th>
        <th
          key="city"
        >
          CITY
        </th>
      </tr>
    </TableHeader>
  </thead>
</table>
`;
